docs: updating homepage to new requirements (#1066)
canonical/rockcraft: docs/index.rst

generation of secure, stable and `OCI-compliant container images
<OCI_image_spec_>`_, based on Ubuntu.

**Rockcraft offers a truly declarative way for building efficient container images.**
By making use of existing Ubuntu tools like LXD_ and Multipass_, Rockcraft
is able to compartmentalise typical container image builds into multiple
parts, each one being comprised of several independent lifecycle steps,
allowing complex operations to be declared at build time.

> Rockcraft enables users to create automatically secure container images in sync with Ubuntu OS’s SLAs,

Reviewer: ```suggestion
Rockcraft enables users to create automatically secure container images in sync with Ubuntu releases,
```

Author: This has been applied.

whilst ensuring the smallest footprint possible.

**Rockcraft is for anyone who wants to build production-grade container images,
regardless of their experience as a software developer** -- from independent
software vendors to cloud-native developers and occasional container users.
Rockcraft handles all the repetitive and boilerplate steps of a build,
directing your focus to what really matters: the image's content.

In this documentation
---------------------

Getting started with Rockcraft
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Understand Rockcraft and build your first Rock.

> * **Tutorial:** :ref:`Build your first rock in 15 min <tutorial-create-a-hello-world-rock>`

Reviewer: ```suggestion
* **Tutorial:** :ref:`Build your first rock <tutorial-create-a-hello-world-rock>`
```

I suggest we don't focus on the duration here. Simplicity is important, but rushing into designing a container isn't what we want for our users.

Author: I have removed it in the newest version.

* **Basic concepts:** :ref:`Lifecycle details of a rock <craft-parts:lifecycle>`

Building rocks
~~~~~~~~~~~~~~

Configure, build, harden, and publish rocks with Rockcraft.

*  **Set up your rock:** :ref:`reference-rockcraft-yaml` •
   :ref:`Types of bases <explanation-bases>` • :ref:`Specify a base <how-to-specify-a-base>` •
   :ref:`Pebble as entrypoint <explanation-pebble>` •
   :ref:`Convert an entrypoint to a Pebble layer <how-to-convert-an-entrypoint-to-a-pebble-layer>` •
   :ref:`Define a non root user <how-to-add-an-internal-user>`

> *  **Hardening:**  :ref:`Overview <explanation-chisel>` • :ref:`Chisel rocks <how-to-chisel-a-rock>` •

Author: Hardening is the industry term users will be familar with.

   :ref:`Install a custom slice <how-to-install-a-custom-package-slice>` •

>    :ref:`Migrate Docker images to chiseled rocks <how-to-migrate-a-docker-image-to-a-chiselled-rock>`

Author: We discussed the wording and formatting offline, the submitted version is what we agreed on.

*  **Pack:** :ref:`Use the Rockcraft pack GitHub Action<how-to-use-the-rockcraft-pack-github-action>` •
   :ref:`Pack a Python package <tutorial-pack-a-python-package>`
* **Publish:** :ref:`Publish a rock to a registry <how-to-publish-a-rock-to-a-registry>`

Rock components
~~~~~~~~~~~~~~~

Define how a rock is built and run.

* **Steps:** :ref:`ref_lifecycle_commands` • :ref:`expl_prime_step_OCI_layer`
* **Parts:** :ref:`Understand Parts <craft-parts:parts>` • :ref:`craft-parts:filesets_explanation` • :ref:`Permissions <reference-parts-and-steps>`
*  **Extensions:** :ref:`Extensions available <reference-extensions>` • :ref:`Commands <ref_extension_commands>` •
   :ref:`configure-the-build-base-for-an-express-app`
*  **Plugins:** :ref:`Plugins available <reference-plugins>` • :ref:`how-to-override-a-plugins-build` •
   :ref:`Include local files and remote resources using the dump plugin <how-to-include-local-files-and-remote-resources>`

How this documentation is organized
-----------------------------------

*  The :ref:`Tutorial <tutorial>` takes you step-by-step through building your first rock.
*  :ref:`How-to guides <how-to>` assume you have basic familiarity with Rockcraft.

>    These include crafting rocks, building rocks for different apps and hardening.

Reviewer: ```suggestion
   These include crafting rocks, building rocks for different apps and optimization.
```

* :ref:`Reference <reference>` provides a guide to the .yaml files, commands and components.
* :ref:`Explanation <explanation>` includes topic overviews, background and context and detailed discussion.

Project and community
---------------------

Rockcraft is a member of the Ubuntu family. It’s an open source project that warmly welcomes
`community contributions <https://documentation.ubuntu.com/project/contributors/>`_.

Support
~~~~~~~

* `Ubuntu Rocks Discourse <https://discourse.ubuntu.com/c/project/rocks/117>`_
* `Rocks Community on Matrix <https://matrix.to/#/#rocks:ubuntu.com>`_

> * `Contribute to the project <https://github.com/canonical/rockcraft>`_

Reviewer: ```suggestion
* `Contribute to the project <https://github.com/canonical/rockcraft/blob/main/CONTRIBUTING.md>`_
```

Author: This has been applied.
